metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java

```diff
@@ -63,9 +64,13 @@ public ResourceExtractionResultImpl copyMetadata(Resource resource, String detec
         resource.getResourceUrl(), resource.getProvidedFileSize()));
   }
 
+  boolean generateThumbnailForPdf(Resource resource, boolean mainThumbnailAvailable) {
+    return !(mainThumbnailAvailable && resource.getUrlTypes().contains(UrlType.IS_SHOWN_BY));
```

Comment:
You mean the check whether the resource is an isShownBy? Yes, we need that. It can be any kind of link (like hasView) and we only skip thumbnail generation in the case of isShownBy links. 
Or am I missing something?

Reply:
Isn't the resource coming in with the mainThumbnailAvailable flag pre-computed?
I mean, we don't need to check if the resource is an isShownBy here because we would already know if we need to process the resource because it comes with the mainThumbnailAvailable.
So if it's mainThumbnailAvailable is true, we don't really need to find the urlTypes here since we already checked that, right? Or not?

Reply:
No, the flag tells us whether a main thumbnail is available. But that doesn't mean we don't need to perform any thumbnail extraction anymore at all. For instance, for hasView resources we still want to extract the thumbnails because their thumbnail is not going to be the 'same' as the main thumbnail. We specifically exclude the isShownBy because that thumbnail is expected to be the 'same' as the main thumbnail.

Reply:
ok